.github/workflows/bump-version-and-create-release.yaml

```diff
@@ -0,0 +1,78 @@
+name: Bump version and create release
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+      - github_action # TODO: Remove before merge
+
+jobs:
+  bump-version:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    outputs:
+      new_tag: ${{ steps.bump_version.outputs.new_tag }}
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v4
+
+      - name: Read version from file
+        run: |
+          # Read the version from the version file, only store the number (without the 'v')
+          INITIAL_VERSION=$(cat version | cut -d'v' -f2)
+          echo "Current version: $INITIAL_VERSION"
+          echo "INITIAL_VERSION=${INITIAL_VERSION}" >> $GITHUB_ENV
+
+      - name: Bump version and push tag
+        id: bump_version
+        uses: anothrNick/github-tag-action@v1
+        env:
+          DEFAULT_BUMP: patch
+          DRY_RUN: true
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          INITIAL_VERSION: ${{ env.INITIAL_VERSION }}
+          WITH_V: true
+
+      - name: Update version file with new version
+        run: |
+          echo "New version: ${{ steps.bump_version.outputs.new_tag }}"
+          echo "VERSION=${{ steps.bump_version.outputs.new_tag }}" > version
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git add version
+          git commit -m "chore: update version file to ${{ steps.bump_version.outputs.new_tag }}"
+          git push
+
+      - name: Push new tag
+        run: |
+          git tag ${{ steps.bump_version.outputs.new_tag }}
+          git push origin ${{ steps.bump_version.outputs.new_tag }}
+
+  create-release:
+    needs: bump-version
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Build release packages
+        uses: docker/build-push-action@v6
+        with:
+          tags: |
+            util_caching_release
+          push: false
+          target: release
+
+      - name: Copy release packages
+        run: |
+          mkdir -p /tmp/artifacts/
+          docker run --rm -v /tmp/artifacts:/tmp/artifacts util_caching_release cp -r /release /tmp/artifacts/
+
+      - name: Create Release
+        uses: ncipollo/release-action@v1
+        with:
+          artifacts: "/tmp/artifacts/release/*"
+          tag: ${{ needs.bump-version.outputs.new_tag }}
+
```

Comment:
It would be nice, if the release description would contain the corresponding PR description.
[Currently](https://github.com/KIT-MRT/util_caching/releases), it's the version bump commit message :see_no_evil: 

I imagine this could work:
- save the merge commit message into a tmp file (must be an early github action step, before version bump commit)
- pass this file to the `ncipollo/release-action` action as `bodyFile`